src/shaders/circle.vertex.glsl

```diff
@@ -80,9 +100,24 @@ void main(void) {
     // multiply a_pos by 0.5, since we had it * 2 in order to sneak
     // in extrusion data
     vec2 circle_center = floor(a_pos * 0.5);
+
+#ifdef PROJECTION_GLOBE_VIEW
+    // Compute positions on both globe and mercator plane to support transition between the two modes
+    // Apply extra scaling to extrusion to cover different pixel space ratios (which is dependant on the latitude)
+    vec2 scaled_extrude = extrude * a_scale;
+    vec3 pos_normal_3 = a_pos_normal_3 / 16384.0;
+    vec3 globe_surface_extrusion = extrudeOnGlobeSurface(scaled_extrude, pos_normal_3, u_up_dir, u_zoom_transition);
+    vec3 globe_elevation = elevationVector(circle_center) * circle_elevation(circle_center);
+    vec3 globe_pos = a_pos_3 + globe_surface_extrusion + globe_elevation;
+    vec3 merc_pos = mercator_tile_position(u_inv_rot_matrix, circle_center, u_tile_id, u_merc_center) + globe_surface_extrusion + globe_elevation;
```

Comment:
Have you checked that elevation vector `globe_elevation` is correct in this case? I believe something like `u_up_dir * u_tile_up_scale * elevation(circle_center)` should be used here as the globe is being unwrapped to a plane that's perpendicular to `u_up_dir`

Reply:
That's interesting because you're right, but it also looked correct when I tested it. I'll check it again!